job-distributor/v1/job/job.proto

```diff
@@ -148,6 +151,22 @@ message ProposeJobResponse {
   Proposal proposal = 1; // Details of the newly created proposal.
 }
 
+// BatchProposeJobRequest contains the job spec and list of nodes to propose it to.
+message BatchProposeJobRequest {
+  repeated string node_ids = 1; // List of node IDs to which the job will be proposed.
+  string spec = 2; // Shared job spec across all nodes.
+  repeated api.label.Label labels = 3; // Shared labels associated with the job.
```

Comment:
Would there be a scenario where user want different labels to different jobs in the batch endpoint? Not that i am saying we want to do it but just checking.

Reply:
I've considered this possibility, but it seems unlikely to occur often, and I’m not sure if we want to enable this functionality. It might end up adding unnecessary complexity. @jkongie do you have any opinion on this?